### PR TITLE
Adds in extra glockroaches.

### PR DESCRIPTION
**Pull request by CRITAWAKETS**
## About The Pull Request

This PR adds in 5 new variants of the glockroach for further admeme customization. The glockroach prime, the glockroach umbra, the magnumroach, the deagleroach and finally the elusive godroach. Be very afraid of central command if this PR is merged.

## Why It's Good For The Game

An admin liked the idea of extra glockroaches, and i just could not resist doing this.

## Changelog
:cl:
add: There are now 5 new special glockroaches! Find and catch em all!
/:cl:

also the glockroach prime and godroach are bugged because from what i can tell, ranged_cooldown is bugged. This is just a meme PR though so fixing that is not really in the scope of this PR. All it results in is a lesser firing speed than it should be.

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -1. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by tralezab** [comment]
Where is south bronx glockroach?

**Comment by Fikou** [comment]
> also the glockroach prime and godroach are bugged because from what i can tell, ranged_cooldown is bugged. This is just a meme PR though so fixing that is not really in the scope of this PR. All it results in is a lesser firing speed than it should be.

yeah guys! my pr doesnt work correctly but its out of scope to fix it...

**Comment by CRITAWAKETS** [comment]
> yeah guys! my pr doesnt work correctly but its out of scope to fix it...

the fix is changing pretty much the entirety of simplemob code to use delta_time and im not good enough at code for that

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm [hunk 1]
```diff
@@ -106,3 +106,107 @@
...
+	melee_damage_lower = 9001
+	melee_damage_upper = 9001
+	obj_damage = 9001
+	gold_core_spawnable = NO_SPAWN
+	projectiletype = /obj/projectile/glockroachbullet/god
+	casingtype = /obj/item/ammo_casing/glockroach/god
+	squish_chance = 0 //oh no
+	health = 13372462 //oh NO
+	maxHealth = 13372462
+	ranged_cooldown_time = 1
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/hostile/cockroach.dm:201]: Did you want the glockroach to fire once every tick?

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm [hunk 2]
```diff
@@ -106,3 +106,107 @@
...
 	return FALSE
+
+/mob/living/simple_animal/hostile/cockroach/glockroach/prime
+	name = "glockroach prime"
+	desc = "The ultimate form of the glockroach. Featuring altered mod polarities that offer greater customization."
+	icon_state = "glockroach-prime"
+	gold_core_spawnable = NO_SPAWN
+	health = 2 //extra armor
+	maxHealth = 2
+	ranged_cooldown_time = 20 //shuuts faster
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/hostile/cockroach.dm:117]: You said this wasn't working, and said something about delta time, could you elaborate further?

**Review by LemonInTheDark — COMMENTED** [review]
Some questions about the memes contained within this pr, I'm not for or against merging it, just want to know what the issue you found with `ranged_cooldown_time` was

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm (hunk 1 above)
Comment (CRITAWAKETS) [reply code/modules/mob/living/simple_animal/hostile/cockroach.dm:201]: Yes.

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm (hunk 2 above)
Comment (CRITAWAKETS) [reply code/modules/mob/living/simple_animal/hostile/cockroach.dm:117]: Basically, ranged_cooldown_time does absolutely nothing. From what i could tell in the hostile mob code, it seems to just be world time being screwy as nothing else seems to be out of place.

**Comment by optimumtact** [comment]
> You currently have a negative Fix/Feature pull request delta of -1. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

+sprite bloat  meme PR

=

gtfo

**Pull request closed without merging**